File: src/main/java/org/normal/api/java/context/ContextType.java
package org.normal.api.java.context;

public record ContextType(String type) {

    public static ContextType fromUser(String type) {
        if (type.isBlank()) throw new IllegalArgumentException("Type cannot be blank");
        String normalized = type.trim().toLowerCase();

        return new ContextType(normalized);
    }

}

File: src/main/java/org/normal/api/kt/context/ContextType.kt
package org.normal.api.kt.context

data class ContextType(val context : String) {

    companion object {
        fun fromUser(context: String) : ContextType {
            require(context.isNotBlank()) { "Context cannot be blank" }
            val normalized = context.trim().lowercase()

            return ContextType(normalized)
        }
    }

}